### DISCUSSION TITLE
Add tenantServices and tenantAppBuilder within a module

### DISCUSSION THREAD

**Discussion by mattyopp** [OrchardCMS/OrchardCore#12119] [Q&A] [answered]
I may not be understanding this properly, but I use Hangfire within an Orchard Core multitenant module. When the module is enabled for a single tenant, Hangfire works properly. As soon as I add a second tenant, Hangfire is confused as to which tenant is triggering the procedures. Is there a way I can ensure Hangfire is isolated for each tenant. 

Any ideas are welcome. One idea I had is to load them like this:
`services.AddOrchardCms()
    .ConfigureServices(tenantServices =>
    {
        ...
    })
    .Configure((tenantAppBuilder, ...) =>
    {
        ...
    });`

However, I'm using a module to load Hangfire. How can I add tenantServices and tenantAppBuilder from within a module?

**Comment by ns8482e** [comment, marked as answer]
Create orchard core module using dot net new and configure in module’s startup.cs

**Reply by mattyopp** [reply]
Thanks. I've taken that step but the Hangfire service is being shared among all of the tenants that have the same module enabled. I load it in the startup.cs like this:
`
public override void ConfigureServices(IServiceCollection services)
{
services.AddHangfire(configuration => configuration
...
`
Is there a way to load the service to be isolated for each tenant?

**Reply by ns8482e** [reply]
HostedService are not tenant aware and shared between all tenants 

use OrchardCore’s BackgroundTask

**Reply by mattyopp** [reply]
Ok thanks. I'll explore other options.
